sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlockBlobAPITest.groovy

```diff
@@ -651,19 +648,20 @@ class BlockBlobAPITest extends APISpec {
             .verifyComplete()
 
         then:
-        def outFile = file.getPath().toString() + "result"
-        def outChannel = AsynchronousFileChannel.open(Paths.get(outFile), StandardOpenOption.CREATE,
-            StandardOpenOption.WRITE)
-        StepVerifier.create(FluxUtil.writeFile(blobAsyncClient.download(), outChannel)).verifyComplete()
-        outChannel.close()
+        def outFile = new File(file.getPath().toString() + "result")
+        outFile.createNewFile()
 
-        compareFiles(file, new File(outFile), 0, fileSize)
+        def outStream = new FileOutputStream(outFile)
+        outStream.write(FluxUtil.collectBytesInByteBufferStream(blobAsyncClient.download()).block())
+        outStream.close()
```

Comment:
This can be simplified by using try-with-resources pattern:

```
new FileOutputStream(outFile).with { outStream ->
      outStream.write(FluxUtil.collectBytesInByteBufferStream(blobAsyncClient.download()).block());
}
```